sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -190,6 +190,7 @@ private[hive] class HiveClientImpl(
     if (clientLoader.cachedHive != null) {
       Hive.set(clientLoader.cachedHive.asInstanceOf[Hive])
     }
+    state.getConf.setClassLoader(initClassLoader)
```

Comment:
We need to add comments to explain this change.

Reply:
Could we just set the `ClassLoader` for Hive 2.3?
```scala
if (HiveUtils.isHive23) {
  state.getConf.setClassLoader(clientLoader.classLoader)
}
```

Reply:
Could you help fix [this incorrect comment](https://github.com/apache/spark/pull/25775/files#diff-6fd847124f8eae45ba2de1cf7d6296feR164): `initialConf.setClassLoader(initClassLoader)` -> `hiveConf.setClassLoader(initClassLoader)`

Reply:
> We need to add comments to explain this change.

Will do,  but 
```
if (HiveUtils.isHive23) {
  state.getConf.setClassLoader(initClassLoader)
}
```
can't work with  version higher then 2.3

Reply:
> Could you help fix [this incorrect comment](https://github.com/apache/spark/pull/25775/files#diff-6fd847124f8eae45ba2de1cf7d6296feR164): `initialConf.setClassLoader(initClassLoader)` -> `hiveConf.setClassLoader(initClassLoader)`

Done